pkg/k8s/logfields.go

```diff
@@ -22,10 +27,10 @@ const (
 	// fieldMaxRetry is the maximum number of retries
 	fieldMaxRetry = "maxRetry"
 
-	// fieldNodeName is the Kubernetes node name where the agent is running on
-	fieldNodeName = "nodeName"
-
-	// fieldSubsys is set to subsysKubernetes on all Kubernetes logging messages
-	fieldSubsys      = "subsys"
+	// subsysKubernetes is the value for logfields.LogSubsys
 	subsysKubernetes = "kubernetes"
 )
+
+var (
+	pkgLog = log.WithField(logfields.LogSubsys, subsysKubernetes)
```

Comment:
Nice! @tgraf FYI ^^^

@raybejjani since you're doing this, you can remove all prefixes `k8s: ` in the messages

Reply:
This is great! What about importing logrus and instead of having pkgLog, have `log = logrus.WithField...` ? Would result in logging invocations being consistent with the rest of the packages being used.

We still have some remaining log messages that use Kubernetes / k8s / K8s:
```
daemon/daemon.go:			log.Info("k8s mode: Allowing localhost to reach local endpoints")
daemon/k8s_watcher.go:				log.Warn("Consider upgrading kubernetes to >=1.7 to enforce NetworkPolicy version 1")
daemon/k8s_watcher.go:			log.WithError(e).Error("k8sError")
daemon/k8s_watcher.go:	log.WithError(e).Error("k8sError")
daemon/k8s_watcher.go:		log.Debug("Detected k8s <1.7, using TPR instead of CRD")
daemon/k8s_watcher.go:		log.Error("Ignoring invalid k8s NetworkPolicy addition")
daemon/k8s_watcher.go:		log.Error("Ignoring invalid k8s NetworkPolicy deletion")
daemon/k8s_watcher.go:		log.Error("Ignoring invalid k8s v1beta1 NetworkPolicy addition")
daemon/k8s_watcher.go:		log.Error("Ignoring invalid k8s v1beta1.NetworkPolicy deletion")
daemon/main.go:		log.WithError(err).Warn("Error while enabling k8s watcher")
pkg/nodeaddress/node_address.go:			log.Warningf("k8s: Can't use CIDR '%s' from kubernetes: %s", node.IPv6AllocCIDR, err)
pkg/workloads/containerd/watcher.go:	log.Debugf("Connecting to kubernetes to retrieve labels for pod %s ns %s", podName, ns)
```

Can you fix these inconsistencies in the above log messages as well?

Reply:
@aanm Done. I removed it from the 2 fmt.Errorf cases too, but I made those messages mention k8s. This makes those 2 consistent with our other errors from this package. I left daemon messages alone (one instance in daemon/daemon.go and some in daemon/k8s_watcher.go) since we expect that code to migrate into pkg/k8s at some point. There was also one in nodeaddress that I left alone but it isn't clearly a k8s specific message so I'm not sure.

@ianvernon This is a fantastic idea! I was using k8s as the testbed for tagging messages with subsystem and what to call the variable was bugging me. Now I know how to do this all over the code ;)
I polled 1 aanm shaped person, and the preference was for `k8s` so I made that how we refer to it in all logs within this package, along with the daemon, containerd watcher and nodeaddress.